Rogue-Ape-Studios/Crusader
# 🐛 Fix PlayerHealth not assigning properly

# Content

The playerhealth did not correctly grab the maxhealth from the script, which caused the system to use the HealthScript base of 100. This has been fixed by moving the implementation out of Health and into the specific scripts fixed the issue.

## Changes


### Updated
`Assets/Scripts/Health/EnemyHealth.cs` : Was updated, now has it's own MaxHealth implementation.
`Assets/Scripts/Health/Health.cs` : Extracted implementation to `PlayerHealth` and `EnemyHealth`.
`Assets/Scripts/Health/PlayerHealth.cs` : Fixed issue where the `ScriptableObject` MaxHealth property did not apply properly.
`Assets/Scripts/UI/HealthBar.cs` : Was updated with the changes to the health system.

## Testing

The feature / Bugfix can be testing by following these steps:

1. Load the level
2. Check the player and enemy health


Closes #239
